### PR TITLE
Fix Issue No: 2945

### LINKED ISSUE #2945: Too much space in the loops and other constructs in the compact view
Release 0.93

compact1

### PR DESCRIPTION
**Pull request by nadeeshaan**
Resolves #2945

### CODE REVIEW

File: modules/web/js/ballerina/diagram/views/compact/dimension-visitors/aborted-statement-dimension-calculator-visitor.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * @class AbortedStatementDimensionCalculatorVisitor
+ * */
+class AbortedStatementDimensionCalculatorVisitor {
+
+    /**
+     * Constructor for Aborted statement dimensions
+     * @param {object} options - options
+     */
+    constructor(options) {
+        this.sizingUtil = new SizingUtil(options);
```
Comment (manuranga) [inline modules/web/js/ballerina/diagram/views/compact/dimension-visitors/aborted-statement-dimension-calculator-visitor.js]: Initializing a util function is not good. Util should be static.

File: modules/web/js/ballerina/diagram/views/compact/dimension-visitors/catch-statement-dimension-calculator-visitor.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+import SizingUtil from './../sizing-util';
+
+/**
+ * Dimension visitor class for catch statement.
+ *
+ * @class CatchStatementDimensionCalculatorVisitor
+ * */
+class CatchStatementDimensionCalculatorVisitor {
```
Comment (manuranga) [inline modules/web/js/ballerina/diagram/views/compact/dimension-visitors/catch-statement-dimension-calculator-visitor.js]: can we extend from a class like CompoundStatementDimensionCalculatorVisitor and have this as an empty class?

File: modules/web/js/ballerina/diagram/views/compact/dimension-visitors/aborted-statement-dimension-calculator-visitor.js (hunk 1 above)
Comment (nadeeshaan) [reply modules/web/js/ballerina/diagram/views/compact/dimension-visitors/aborted-statement-dimension-calculator-visitor.js]: This is the first phase of iterative migration of the diagram rendering logic to perspective based rendering. In the current approach, this particular util needs to be configured based on the designer (merge of default and the perspective's designer). With the current implementation, we need to initialize the util.

File: modules/web/js/ballerina/diagram/views/compact/dimension-visitors/catch-statement-dimension-calculator-visitor.js (hunk 2 above)
Comment (nadeeshaan) [reply modules/web/js/ballerina/diagram/views/compact/dimension-visitors/catch-statement-dimension-calculator-visitor.js]: This requires a major refactoring and will be focused in the future release

**Pull request merged**